src/NUnitEngine/nunit.engine/Drivers/NUnit3FrameworkDriver.cs

```diff
@@ -120,7 +121,24 @@ public string Run(ITestEventListener listener, string filter)
             CallbackHandler handler = new RunTestsCallbackHandler(listener);
 
             log.Info("Running {0} - see separate log file", Path.GetFileName(_testAssemblyPath));
-            CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+
+            try
+            {
+                CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+            }
+            catch (System.Reflection.TargetInvocationException ex)
+            {
+                string innerMessage = ex.InnerException.Message;
+                if (ex.InnerException is ArgumentException && innerMessage.StartsWith("Invalid filter element"))
```

Comment:
The driver currently knows nothing whatsoever about filters by design. In fact, it knows nothing about any of the content of the xml and exceptions that pass through it. That's what keeps the driver maintainable.

Additionally, consider that the engine may use some other driver, like the V2 driver. It won't understand the filter either but some kind of an error will propagate back to the engine and possibly to the runner.